src/dist/replication/test/meta_test/unit_test/meta_duplication_service_test.cpp

```diff
@@ -650,5 +651,34 @@ TEST_F(meta_duplication_service_test, recover_from_corrupted_meta_data)
     test_recover_from_corrupted_meta_data();
 }
 
+TEST_F(meta_duplication_service_test, query_duplication_handler)
+{
+    std::string test_app = "test-app";
+    create_app(test_app);
+    create_dup(test_app);
+    meta_http_service mhs(_ms.get());
+
+    http_request fake_req;
+    http_response fake_resp;
+    fake_req.query_args["name"] = test_app + "not-found";
+    mhs.query_duplication_handler(fake_req, fake_resp);
+    ASSERT_EQ(fake_resp.status_code, http_status_code::not_found);
+
+    const auto &duplications = find_app(test_app)->duplications;
+    ASSERT_EQ(duplications.size(), 1);
+    auto dup = duplications.begin()->second;
+
+    fake_req.query_args["name"] = test_app;
+    mhs.query_duplication_handler(fake_req, fake_resp);
+    ASSERT_EQ(fake_resp.status_code, http_status_code::ok);
+    ASSERT_EQ(
+        fake_resp.body,
+        std::string("{\"") + std::to_string(dup->id) + "\":{\"create_ts\":" +
+            std::to_string(dup->create_timestamp_ms) +
+            ",\"progress\":{\"0\":-1,\"1\":-1,\"2\":-1,\"3\":-1,"
```

Comment:
Using https://en.cppreference.com/w/cpp/language/string_literal is more readable